src/components/TopBar/AboutDialog.js

```diff
@@ -0,0 +1,359 @@
+    Dialog,
+    DialogActions,
+    DialogContent,
+    DialogTitle,
+    Fade,
+    Grid,
+    Skeleton,
+    Tooltip,
+    useMediaQuery,
+    useTheme,
+} from '@mui/material';
+import {
+    Bookmark,
+    BrokenImage,
+    DnsOutlined,
+    Gavel,
+    QuestionMark,
+    WidgetsOutlined,
+} from '@mui/icons-material';
+import { FormattedMessage } from 'react-intl';
+import PropTypes from 'prop-types';
+
+const ComponentsTypeIcons = {
+    app: <WidgetsOutlined />,
+    server: <DnsOutlined />,
+    other: <QuestionMark />,
+};
+
+const AboutDialog = ({
+    open,
+    onClose,
+    getGlobalVersion,
+    getLogoThemed,
+    appVersion,
+    appLicense,
+    getAdditionalComponents,
+}) => {
+    const theme = useTheme();
+
+    //TODO is useCallback in component or in caller?
+    const logo = useMemo(() => {
+        if (getLogoThemed) {
+            return getLogoThemed(theme.palette.mode);
+        } else {
+            return <BrokenImage />;
+        }
+    }, [getLogoThemed, theme.palette.mode]);
+
+    const handlerGetGlobalVersion = useCallback(getGlobalVersion, [
+        getGlobalVersion,
+    ]);
+    const [loadingGlobalVersion, setLoadingGlobalVersion] = useState(false);
+
+    /* We want to get the initial version once at first render to detect later a new deploy */
+    const [startingGlobalVersion, setStartingGlobalVersion] = useState(null);
+    useEffect(() => {
+        if (getGlobalVersion) {
+            handlerGetGlobalVersion((value) => {
+                setLoadingGlobalVersion(false);
+                setStartingGlobalVersion(value);
+                setActualGlobalVersion(value);
+            });
+        }
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, []);
+
+    const [actualGlobalVersion, setActualGlobalVersion] = useState(null);
+    useEffect(() => {
+        if (open && getGlobalVersion) {
+            setLoadingGlobalVersion(true);
+            handlerGetGlobalVersion((value) => {
+                setLoadingGlobalVersion(false);
+                setActualGlobalVersion(value);
+            });
+        }
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, [open, handlerGetGlobalVersion]);
+
+    const [loadingAdditionalComponents, setLoadingAdditionalComponents] =
+        useState(false);
+    const [additionalComponents, setAdditionalComponents] = useState(null);
+    useEffect(() => {
+        if (open && getAdditionalComponents) {
+            setLoadingAdditionalComponents(true);
+            getAdditionalComponents((values) => {
+                setLoadingAdditionalComponents(false);
+                setAdditionalComponents(values);
+            });
+        }
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, [open, getAdditionalComponents]);
+
+    const handleClose = useCallback(() => {
+        if (onClose) {
+            onClose();
+        }
+        setAdditionalComponents(null);
+        setActualGlobalVersion(null);
```

Comment:
These 2 lines are necessary ?

Reply:
yes to not keep old values and having them shown next time the dialog is reopen before data are reloaded

Reply:
Ah yes we use an `open` prop and we don't unmount the component

Reply:
yes to keep the global version in memory